src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -980,6 +980,43 @@ Example:
     def list(self):
       return {"msg": "Hello"}
 
+How to create a dedicated UI endpoint which uses the 'public' API?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes we want to combine multiple calls into one single call
+to save bandwidth or for other performance reasons.
+In order to achieve that, we first have to create an ``@UiApiController`` which
+is used for endpoints consumed by the UI but that are not part of the
+'public' API.
+During ``__init__`` we initialize the ``@ApiController``s that we want to use
+and attach the resulting objects to ``self``.
+
+Example:
+
+.. code-block:: python
+
+  import cherrypy
+  from . import UiApiController, ApiController, RESTController
+
+
+  @ApiController('ping', secure=False)  # /api/ping
+  class Ping(RESTController):
+    def list(self):
+      return self._list
+
+    def _list(self):  # To not get in conflict with the JSON wrapper
+      return {"msg": "Hello"}
```

Comment:
Feels kind of weird to return a dict in the list method. Can you change this to return a list object with something inside, and in the `PingUi` class append more values to it?